catroid/src/main/java/org/catrobat/catroid/nfc/NfcHandler.java

```diff
@@ -83,10 +135,119 @@ public static String byteArrayToHex(byte[] a) {
 		for (byte b : a) {
 			sb.append(String.format("%02x", b & 0xff));
 		}
-		return cutHexStringToIntSize(sb.toString());
+		return sb.toString();
 	}
 
-	public static String cutHexStringToIntSize(String longHex) {
-		return String.valueOf(Long.parseLong(longHex, 16) & Integer.MAX_VALUE);
+	public static void writeTag(Tag tag, NdefMessage message) {
+		if (tag != null) {
+			try {
+				Ndef ndefTag = Ndef.get(tag);
+				if (ndefTag == null) {
+					NdefFormatable nForm = NdefFormatable.get(tag);
+					if (nForm != null) {
+						nForm.connect();
+						nForm.format(message);
+						nForm.close();
+					}
+				} else {
+					ndefTag.connect();
+					ndefTag.writeNdefMessage(message);
+					ndefTag.close();
+				}
+			} catch (Exception e) {
```

Comment:
[Do not catch Exception](http://checkstyle.sourceforge.net/config_coding.html#IllegalCatch)